menas/src/test/scala/za/co/absa/enceladus/menas/integration/repositories/StatisticsIntegrationSuite.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.menas.integration.repositories
+
+import org.junit.runner.RunWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.junit4.SpringRunner
+import za.co.absa.enceladus.menas.integration.fixtures.{DatasetFixtureService, FixtureService, PropertyDefinitionFixtureService}
+import za.co.absa.enceladus.menas.repositories.{DatasetMongoRepository, PropertyDefinitionMongoRepository}
+import za.co.absa.enceladus.menas.services.StatisticsService
+import za.co.absa.enceladus.model.properties.{PropertyDefinition, PropertyDefinitionDto}
+import za.co.absa.enceladus.model.properties.essentiality.Essentiality.{Mandatory, Optional, Recommended}
+import za.co.absa.enceladus.model.properties.propertyType.{EnumPropertyType, StringPropertyType}
+import za.co.absa.enceladus.model.test.factories.DatasetFactory
+
+@RunWith(classOf[SpringRunner])
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles(Array("withEmbeddedMongo"))
+class StatisticsIntegrationSuite extends BaseRepositoryTest {
+  @Autowired
+  private val datasetFixture: DatasetFixtureService = null
+
+  @Autowired
+  private val propertyDefService: PropertyDefinitionFixtureService = null
+
+  @Autowired
+  private val datasetMongoRepository: DatasetMongoRepository = null
+
+  @Autowired
+  private val propertyDefinitionRepository: PropertyDefinitionMongoRepository = null
+
+  @Autowired
+  private val statisticsService: StatisticsService = null
+
+  override def fixtures: List[FixtureService[_]] = List(datasetFixture)
+
+  val mockedDatasets = Seq(
```

Comment:
If I am not mistaken, there is no dataset that having a property value present in a non-latest version while having the same property value missing in the latest version (eg. 
```
version1: recommendedString1= some value 
version2: (recommendedString1 unset)
```
)

This is a real situation (non-mandatory properties can be unset) and the test should cover this case, too, I believe -- to correctly report the property as missing (bc. of evaluating the latest version only).

Reply:
Good point